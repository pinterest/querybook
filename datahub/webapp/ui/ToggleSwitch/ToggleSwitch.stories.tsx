import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { ToggleSwitch } from './ToggleSwitch';

export const _ToggleSwitch = () => {
    const [checked, setChecked] = React.useState(false);
    const onChange = () => {
        setChecked(!checked);
    };
    return <ToggleSwitch checked={checked} onChange={onChange} />;
};

export default {
    title: 'Form/ToggleSwitch',
    decorators: [centered],
};
