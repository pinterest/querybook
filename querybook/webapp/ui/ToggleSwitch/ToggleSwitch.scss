.ToggleSwitch {
    position: relative;
    display: inline-block;
    width: 48px;
    height: 24px;
    border-radius: 12px;
    background-color: var(--color-null);
    cursor: pointer;
    transition: all 0.25s ease-out;

    .ToggleSwitch-circle {
        position: absolute;
        top: 3px;
        left: 3px;
        width: 18px;
        height: 18px;
        background-color: var(--bg-color);
        border-radius: 50%;
        transition: all 0.25s ease-out;
    }

    &.checked {
        background-color: var(--color-accent-bg);
        .ToggleSwitch-circle {
            left: 25px;
        }
    }
}
