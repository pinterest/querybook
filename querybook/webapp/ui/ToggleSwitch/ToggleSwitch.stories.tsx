import centered from '@storybook/addon-centered/react';
import React from 'react';

import { ToggleSwitch } from './ToggleSwitch';

export const _ToggleSwitch = () => {
    const [checked, setChecked] = React.useState(false);
    const onChange = () => {
        setChecked(!checked);
    };
    return <ToggleSwitch checked={checked} onChange={onChange} />;
};

export default {
    title: 'Form/ToggleSwitch',
    decorators: [centered],
};
