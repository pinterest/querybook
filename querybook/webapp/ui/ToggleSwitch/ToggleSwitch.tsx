import React from 'react';

import './ToggleSwitch.scss';

export interface IToggleSwitchProps {
    onChange: (checked: boolean) => any;
    checked: boolean;
}

export const ToggleSwitch: React.FunctionComponent<IToggleSwitchProps> = ({
    onChange,
    checked,
}) => (
    <div
        className={`ToggleSwitch ${checked ? 'checked' : ''}`}
        onClick={() => onChange(!checked)}
    >
        <span className="ToggleSwitch-circle" />
    </div>
);
