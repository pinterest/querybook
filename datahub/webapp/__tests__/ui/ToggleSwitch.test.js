import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ToggleSwitch } from '../../ui/ToggleSwitch/ToggleSwitch';

it('renders without crashing', () => {
    shallow(<ToggleSwitch checked={true} onChange={() => null} />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        let wrapper = shallow(
            <ToggleSwitch checked={true} onChange={() => null} />
        );
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
    it('matches snapshot - unchecked', () => {
        let wrapper = shallow(
            <ToggleSwitch checked={false} onChange={() => null} />
        );
        let serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});
