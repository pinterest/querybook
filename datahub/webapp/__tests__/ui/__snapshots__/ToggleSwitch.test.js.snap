// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot - unchecked 1`] = `
<div
  className="ToggleSwitch "
  onClick={[Function]}
>
  <span
    className="ToggleSwitch-circle"
  />
</div>
`;

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="ToggleSwitch checked"
  onClick={[Function]}
>
  <span
    className="ToggleSwitch-circle"
  />
</div>
`;
